bootloader/lk.go

```diff
@@ -65,37 +85,158 @@ func (l *lk) Name() string {
 
 func (l *lk) dir() string {
 	// we have two scenarios, image building and runtime
-	// during image building we store environment into file
-	// at runtime environment is written directly into dedicated partition
+	// during image building we store environment into file, while at runtime
+	// the environment is written directly into dedicated partition
 	if l.inRuntimeMode {
```

Comment:
Wondering why this isn't called `preparImageMode` or something similar. Given that prepare image time handling is shorter, the code using that could be refactor to handle that mode first, followed by runtime (!image time). Otherwise it just looks confusing, where code that runs most often is in some `if` branch, looking as if it's some exception/special case.

Reply:
that's a fair point, I think the thinking was that we wanted to make the more "important" and complex case of runtime handling more clear, but I think you're right in that it would actually be simpler to write if we had `if l.prepareImageMode`, etc.

Reply:
this is done